server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -240,11 +245,13 @@ public AppenderatorAddResult add(
     final int numAddedRows = sinkRowsInMemoryAfterAdd - sinkRowsInMemoryBeforeAdd;
     rowsCurrentlyInMemory.addAndGet(numAddedRows);
     totalRows.addAndGet(numAddedRows);
+    currentBytesInMemory.addAndGet(bytesInMemoryAfterAdd - bytesInMemoryBeforeAdd);
 
     boolean isPersistRequired = false;
     if (!sink.canAppendRow()
         || System.currentTimeMillis() > nextFlush
-        || rowsCurrentlyInMemory.get() >= tuningConfig.getMaxRowsInMemory()) {
+        || rowsCurrentlyInMemory.get() >= tuningConfig.getMaxRowsInMemory()
+        || (tuningConfig.getMaxBytesInMemory() > 0 && currentBytesInMemory.get() >= tuningConfig.getMaxBytesInMemory())) {
       if (allowIncrementalPersists) {
         // persistAll clears rowsCurrentlyInMemory, no need to update it.
         persistAll(committerSupplier == null ? null : committerSupplier.get());
```

Comment:
Please log the reason we're persisting here. There are starting to be enough conditions that it's going to be useful to see the specific one that got triggered.

Reply:
Refactored this part a bit, please check the format of the log, if it looks okay.

Reply:
It looks good, although in the nittiest of nit picks, `", "` would be nicer than `","`.